### DISCUSSION TITLE
How much storage is needed?

### DISCUSSION THREAD

**Discussion by jwmurray** [gomods/athens#1678] [General] [answered]
How much storage are people typically allocating or using with Athens?  I would be using it for air-gap development offline.

Thank you

John

**Comment by arschles** [comment, marked as answer]
Hi @jwmurray, that depends on how many dependencies you need to store. Athens runs `go mod download` and stores the results (`go.mod` file, `vX.Y.X.info` file, and the `vX.Y.Z.zip` file), so you can calculate exactly what it will store for an individual module by running that on your machine and adding up those file sizes.

**Comment by jwmurray** [comment]
So, if I want to use Athens as a server for offline development, I have to
know ahead of time the modules  I will want to archive.  Does Athens
provide a means of recording missing modules to be retrieved?

Suppose I attempt a build in my airgap network and find that I am missing
modulex.version6.  I then need to go to my network server, issue the go.mod
file modulex.version6 command and then create a DVD and transfer that to my
airgap directory/Athens server.  Is that easy for the two Athens boxes to
figure out which files are needed for updates?

Thank you!

john

On Thu, Nov 19, 2020 at 1:38 PM Aaron Schlesinger <notifications@github.com>
wrote:

> Hi @jwmurray <https://github.com/jwmurray>, that depends on how many
> dependencies you need to store. Athens runs go mod download and stores
> the results (go.mod file, vX.Y.X.info file, and the vX.Y.Z.zip file), so
> you can calculate exactly what it will store for an individual module by
> running that on your machine and adding up those file sizes.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gomods/athens/discussions/1678#discussioncomment-134243>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAB3HTOZJJBASFCIACEQZBLSQV63XANCNFSM4TY2W6GQ>
> .
>


-- 
John Murray
john@gardenway.org
text:  801-628-0543

**Reply by arschles** [reply]
Athens doesn't formally record what modules it _doesn't_ have, although you could parse the logs to figure it out. It's possible to load Athens with dependencies without requiring network access. We have two resources for doing that. First, here's a [guide](https://docs.gomods.io/configuration/prefill-disk-cache/) on how to pre-fill disk storage. As the title says, it only works with disk-based storage. If you're using other storage backends, we don't have an official solution yet, but there is a work in progress PR published that builds a CLI to support others: https://github.com/gomods/athens/pull/1609.
